src/stacks-hierarchy/Root_PurchaseOverviewScreen/components/TravelDate/TravelDateSheet.tsx

```diff
@@ -29,15 +30,53 @@ type Props = {
   close: () => void;
   save: (dateString?: string) => void;
   maximumDate?: Date;
+  showActivationDateWarning?: boolean;
+  setShowActivationDateWarning: (value: boolean) => void;
 };
 
 export const TravelDateSheet = forwardRef<ScrollView, Props>(
-  ({travelDate, close, save, maximumDate}, focusRef) => {
+  (
+    {
+      travelDate,
+      close,
+      save,
+      maximumDate,
+      showActivationDateWarning,
+      setShowActivationDateWarning,
+    },
+    focusRef,
+  ) => {
     const {t, language} = useTranslation();
     const styles = useStyles();
 
     const defaultDate = travelDate ?? new Date().toISOString();
     const [dateString, setDate] = useState(defaultDate);
+    const [
+      replicatedShowActivationDateWarning,
+      setReplicatedShowActivationDateWarning,
+    ] = useState<boolean>(false);
+
+    const setInternalAndExternalWarningState = (value: boolean) => {
+      setShowActivationDateWarning(value);
+      setReplicatedShowActivationDateWarning(value);
+    };
+
+    const onSetDate = (date: string) => {
+      if (!maximumDate) setDate(date);
+      else {
+        if (isAfter(date, maximumDate)) {
+          setInternalAndExternalWarningState(true);
+        } else if (
+          replicatedShowActivationDateWarning ||
+          showActivationDateWarning
```

Comment:
When replicatedShowActivationDateWarning is initted to showActivationDateWarning, `|| showActivationDateWarning` should not be needed here